syntax = "proto3";

option java_multiple_files = true;
option java_package = "sr.grpc.gen";
option java_outer_classname = "StreamingCOVID";

package streaming;

  
  enum District {
      ALL = 0;
      DOLNOSLASKE = 1;
      KUJAWSKO_POMORSKIE = 2;
      LUBELSKIE = 3;
      LUBUSKIE = 4;	
      LODZKIE = 5;
      MALOPOLSKIE = 6;
      MAZOWIECKIE = 7;
      OPOLSKIE = 8;
      PODKARPACKIE = 9;
      PODLASKIE = 10;
      POMORSKIE = 11;
      SLASKIE = 12;
      SWIETOKRZYSKIE = 13;
      WARMINSKO_MAZURSKIE = 14;
      WIELKOPOLSKIE = 15;
      ZACHODNIOPOMORSKIE = 16;
      POLAND = 17;
   }
message Task{
    District district = 1;
}
message Report
{
    District district = 1;
    int32 died = 2;
    int32 ill = 3;
    int32 quarantine = 4;
}
message Ping_msg
{
    
}
service StreamTester {
  rpc GetNewInfo(Task) returns (stream Report);
  rpc Ping(Ping_msg) returns (Ping_msg);
}
